Bottom: 04fe6dea19880b1b9c863cf66659a7a84dcae1ac
Top:    b31c16549d2d5f34412f7266d7c3254eab3dd876
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-10 14:29:57 +0100

search.c: Don't allow selecting pads or pins of locked elements

The locked flag test was mistakenly being applied ot the
pad or pin its-self, not the parent element.

---

diff --git a/src/search.c b/src/search.c
index 3fafcd4..30fe883 100644
--- a/src/search.c
+++ b/src/search.c
@@ -114,13 +114,14 @@ pinorvia_callback (const BoxType * box, void *cl)
 {
   struct ans_info *i = (struct ans_info *) cl;
   PinTypePtr pin = (PinTypePtr) box;
+  AnyObjectType *ptr1 = pin->Element ? pin->Element : pin;
 
-  if (TEST_FLAG (i->locked, pin))
+  if (TEST_FLAG (i->locked, ptr1))
     return 0;
 
   if (!IsPointOnPin (PosX, PosY, SearchRadius, pin))
     return 0;
-  *i->ptr1 = pin->Element ? pin->Element : pin;
+  *i->ptr1 = ptr1;
   *i->ptr2 = *i->ptr3 = pin;
   longjmp (i->env, 1);
   return 1;			/* never reached */
@@ -181,15 +182,16 @@ pad_callback (const BoxType * b, void *cl)
 {
   PadTypePtr pad = (PadTypePtr) b;
   struct ans_info *i = (struct ans_info *) cl;
+  AnyObjectType *ptr1 = pad->Element;
 
-  if (TEST_FLAG (i->locked, pad))
+  if (TEST_FLAG (i->locked, ptr1))
     return 0;
 
   if (FRONT (pad) || i->BackToo)
     {
       if (IsPointInPad (PosX, PosY, SearchRadius, pad))
 	    {
-	      *i->ptr1 = pad->Element;
+	      *i->ptr1 = ptr1;
 	      *i->ptr2 = *i->ptr3 = pad;
 	      longjmp (i->env, 1);
 	    }
